### PR TITLE
fix(braintree): harden error propagation and document payment provider

### PR DESCRIPTION
**Pull request by SGFGOV**
## Summary
- Document Braintree payment provider helpers and lifecycle methods with JSDoc (current behavior, including throws).
- Stop silently discarding errors in `braintree-base.ts`: webhook parse failures now throw; authorize/account-holder paths preserve typed `MedusaError` via `rethrowGatewayError` / `MedusaError.isMedusaError`; orphan-sale void failures no longer replace the original sync error.
- Align refund unit tests with the `braintreeRefund` history-array shape and add coverage for webhook parse propagation.
- Ignore local `references/` clones in `.gitignore`.

## Test plan
- [x] `npm test -- --testPathPattern='braintree-base|braintree-import'` in `plugins/braintree-payment` (30 passed)
- [ ] Authorize with a declined/validation sale and confirm `PAYMENT_AUTHORIZATION_ERROR` reaches the cart completion flow
- [ ] Submit an invalid Braintree webhook signature and confirm the provider throws (does not return `FAILED`)
- [ ] Refund authorized vs settled transactions and confirm `data.braintreeRefund` history entries (`type` + `transaction`)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved Braintree payment authorization errors with clearer, more consistent validation messages.
  * Refined refund processing across transaction statuses, including sandbox settlement and refund-history handling.
  * Improved webhook handling for unsupported events, missing transaction details, and invalid data.
  * Enhanced reliability for payment deletion and account-holder operations by preserving meaningful error information.
* **Tests**
  * Expanded coverage for refunds, authorization failures, missing transaction IDs, and webhook edge cases.
* **Chores**
  * Excluded `references/` from version control.

### CODE REVIEW

File: plugins/braintree-payment/src/providers/payment-braintree/src/core/braintree-base.ts [hunk 1]
```diff
@@ -793,129 +1107,171 @@ class BraintreeBase extends AbstractPaymentProvider<BraintreeOptions> {
...
+   * @param input - Original refund input (prior history read from `data.braintreeRefund`)
+   * @param transaction - Pre-refund Braintree transaction retained on session data
+   * @param entry - New void/refund history entry
+   */
+  private buildRefundPaymentOutput(
+    input: RefundPaymentInput,
+    transaction: Transaction,
+    entry: BraintreeRefundHistoryEntry,
+  ): RefundPaymentOutput {
+    const stored = input.data?.braintreeRefund;
```
Comment (lcmohsen) [inline plugins/braintree-payment/src/providers/payment-braintree/src/core/braintree-base.ts]: **Bug / regression risk:** 0.1.8 stored refund history under `braintreeRefunds[]` (see CHANGELOG), but this helper now reads/writes `braintreeRefund`.

For sessions that already have partial refunds under `braintreeRefunds`, the next refund will treat prior history as empty, append only the new entry under `braintreeRefund`, and leave the old `braintreeRefunds` array untouched via `...input.data`. History can be lost / both keys can drift.

Prefer keeping `braintreeRefunds` for continuity with 0.1.8, or if renaming intentionally, read from both keys (prefer `braintreeRefunds`) and migrate onto one key when writing. Also update CHANGELOG/tests to match the chosen key.
